libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Util/SFSDKWebUtils.m

```diff
@@ -51,37 +40,4 @@ + (void)configureUserAgent:(NSString *)userAgentString
     }
 }
 
-+ (NSString *)currentUserAgentForApp
-{
-    [self stageUserAgentForApp];
-    
-    return gUserAgentForApp;
-}
-
-+ (void)stageUserAgentForApp
-{
-    if (![NSThread isMainThread]) {
-        dispatch_sync(dispatch_get_main_queue(), ^{
-            [SFSDKWebUtils stageUserAgentForApp];
-        });
-        return;
-    }
-    
-    if (gUserAgentForApp != nil) return;
-
-    __block BOOL finished = NO;
-    
-    WKWebView *webView = [[WKWebView alloc] initWithFrame:CGRectZero];
-
-   [webView evaluateJavaScript:@"navigator.userAgent"
-             completionHandler:^(id _Nullable val, NSError * _Nullable error) {
-                 gUserAgentForApp = val;
-                 finished = true;
-             }];
-
-   while( !finished ) {
```

Comment:
Hackish approach alert! This flat out breaks on `iOS 10`. There's no good way to fetch user agent on `iOS` natively though.